lib/globus_client/endpoint.rb

```diff
@@ -5,6 +5,8 @@ class GlobusClient
   class Endpoint
     PATH_SEPARATOR = "/"
 
+    FileInfo = Struct.new(:name, :size)
```

Comment:
This used to be a hash, but structs are just as fast and give us the ability to make use of short-hand syntax such as `sum(&:size)` over `sum { |file_info| file_info['size'] }` and get us out of the business of caring about strings or symbols as keys and back to message-passing.